components/eam/src/physics/crm/crm_history.F90

```diff
@@ -572,55 +579,56 @@ subroutine crm_history_out(state, ptend, crm_state, crm_rad, crm_output, crm_ecp
    tgliqwp(:ncol) = 0.
    do k = 1,pver
       do i = 1,ncol
-         cicewp(i,k) = crm_output%gicewp(i,k) * 1.0e-3 / max(0.01_r8,crm_output%cld(i,k)) ! In-cloud ice water path.  g/m2 --> kg/m2
-         cliqwp(i,k) = crm_output%gliqwp(i,k) * 1.0e-3 / max(0.01_r8,crm_output%cld(i,k)) ! In-cloud liquid water path. g/m2 --> kg/m2
-         tgicewp(i)  = tgicewp(i) + crm_output%gicewp(i,k) *1.0e-3 ! grid cell mean ice water path.  g/m2 --> kg/m2
-         tgliqwp(i)  = tgliqwp(i) + crm_output%gliqwp(i,k) *1.0e-3 ! grid cell mean ice water path.  g/m2 --> kg/m2
+         icrm = icrm_beg - 1 + i
+         cicewp(i,k) = crm_output%gicewp(icrm,k) * 1.0e-3 / max(0.01_r8,crm_output%cld(icrm,k)) ! In-cloud ice water path.  g/m2 --> kg/m2
+         cliqwp(i,k) = crm_output%gliqwp(icrm,k) * 1.0e-3 / max(0.01_r8,crm_output%cld(icrm,k)) ! In-cloud liquid water path. g/m2 --> kg/m2
+         tgicewp(i)  = tgicewp(i) + crm_output%gicewp(icrm,k) *1.0e-3 ! grid cell mean ice water path.  g/m2 --> kg/m2
+         tgliqwp(i)  = tgliqwp(i) + crm_output%gliqwp(icrm,k) *1.0e-3 ! grid cell mean ice water path.  g/m2 --> kg/m2
       end do
    end do
    tgwp(:ncol) = tgicewp(:ncol) + tgliqwp(:ncol)
-   gwp(:ncol,:pver) = crm_output%gicewp(:ncol,:pver) + crm_output%gliqwp(:ncol,:pver)
+   gwp(:ncol,:pver) = crm_output%gicewp(icrm_beg:icrm_end,:pver) + crm_output%gliqwp(icrm_beg:icrm_end,:pver)
    cwp(:ncol,:pver) = cicewp(:ncol,:pver) + cliqwp(:ncol,:pver)
 
-   call outfld('GCLDLWP' ,gwp,     pcols, lchnk)
-   call outfld('TGCLDCWP',tgwp,    pcols, lchnk)
-   call outfld('TGCLDLWP',tgliqwp, pcols, lchnk)
-   call outfld('TGCLDIWP',tgicewp, pcols, lchnk)
-   call outfld('ICLDTWP' ,cwp,     pcols, lchnk)
-   call outfld('ICLDIWP' ,cicewp,  pcols, lchnk)
+   call outfld('GCLDLWP' ,gwp,     ncol, lchnk)
```

Comment:
Same comment here as above. Don’t these arrays have size pcols, but you’re passing ncol to outfld? I am pretty sure we need to be consistent.

Reply:
I guess in this case we could use pcols... but I'd like to be consistent in our use of "ncol" in this whole routine rather than pcols.

Reply:
To be clear, I don’t want to use pcols…I want ncol, but we also need to make sure the arrays being passed to the routine are only sized ncol and not pcols but with the ncol argument.